### ISSUE TITLE
New version just stopped working

### ISSUE THREAD

**Issue by Jarskaa** [andy-portmen/external-application-button#29]
New version not open popup menu, only default action at extension bar work (open on Edge).

**Comment by andy-portmen** [comment]
Have you tried to restart the browser once? What is your browser and what is your OS?

**Comment by athost** [comment]
Сonfirm, the same issue.
After the last update, the context menu items disappeared.
Linux x86_64, Vivaldi

**Comment by yoyoma2** [comment]
Same here.  The new version doesn't add a menu item to the context menu so it's 100% broken for me.  Firefox on linux.

**Comment by Jarskaa** [comment]
> Have you tried to restart the browser once? What is your browser and what is your OS?
Sorry for my quick post without properly information.

Yes I have, even restart PC for other reasons, without luck. And this is in Windows 10 x64, with Vivaldi browser. Before update context menu working normally, now it have disappeared. Clicking extension icon run default action, on my config "open current page in Edge", but I have quite many context menu operations that do not work right now.

**Comment by yoyoma2** [comment]
Have also rebooted with no change.  The **only** checkbox selected in the addon's preferences is "Selection Context" and since the updated version, the "Display Name" text no longer appears in the selected text right button popup menu.

**Comment by andy-portmen** [comment]
Seems like this is Vivaldi's only issue. Do you guys have any extension that can display the context menu items on the latest release of Vivaldi? I tested a few and none displayed their context menu items.

@yoyoma2 I just tested the extension on a clean FF profile. The context menu items do work. Please use the browser console to see if you get any relevant logs that can help me debug the issue. Alternatively, test on a clean profile.

**Comment by Jarskaa** [comment]
> Seems like this is Vivaldi's only issue. Do you guys have any extension that can display the context menu items on the latest release of Vivaldi? I tested a few and none displayed their context menu items.

I have many, and all of them show context menu on Vivaldi, except External Application Button.

**Comment by andy-portmen** [comment]
@Jarskaa examples for testing please

**Comment by andy-portmen** [comment]
@Jarskaa, for instance, install uBlock Origin and you should have a custom item in the right-click context menu over the toolbar button. Do you have it?

**Comment by Jarskaa** [comment]
http://images.devs-on.net/Image/j9dcIpw2ty6UCzQQ-Wholescreen.png

**Comment by yoyoma2** [comment]
Two linux firefox 69.02 systems had this problem occur with the new version 0.3.4.  The update happened automatically on both and noticed a yellow triangle asking to confirm new permissions.  Why new permissions BTW?

A seldomly used windows 10 firefox 69.03 was still on 0.3.3 and still working.  Forced upgrade to 0.3.4 and now it's broken too.
![eab2](https://user-images.githubusercontent.com/8683106/66924626-f2720f80-eff8-11e9-9f1b-e686574acf5d.png)
![eab1](https://user-images.githubusercontent.com/8683106/66924628-f3a33c80-eff8-11e9-82e4-72858d460c83.png)

**Comment by dbkeitel** [comment]
Firefox 64bit Linux user here. I experienced the same problem with 0.3.4 but it seems to be solvable by the good old "change settings, restart, change back, restart" trick.

For context, I have always used this extension through a toolbar button, switching target applications per right-click on that button, and never through the in-page context menu; so I can't say if the latter used to work before. But right-clicking the toolbar button has been broken for me with 0.3.4 (only showing the default "Manage Extension, "Remove Extension", "Report Extension" entries and _not_ the application switching entries). And at least after that update, when I checked, there was no in-page context menu entry at all.

Interestingly, after unchecking the "Toolbar button" checkbox in the extension preferences for each application, hitting "Update Application" and then restarting Firefox, the in-page context menu entry appered.

Then re-checking the "Toolbar button" checkboxes, hitting "Update Application" and restarting again, switching via the button context menu works again, too!


**Great** extension by the way in general! My favourite use is to combine it with KDE Connect for local-network inter-device sharing.

**Comment by yoyoma2** [comment]
Thanks dbkeitel, your workaround worked.  One machine had lost its settings so on that one had to re-enter and click "Add application" instead of "Update Application".

**Comment by Jarskaa** [comment]
Yep, I only done that update all setting and context menu appeared again immediately (no need to close/restart or so). Thank you guys! :)

**Comment by olliebean** [comment]
Same issue here, in Firefox - restarting the browser didn't fix it, nor did uninstalling the extension and reinstalling v0.3.3. But toggling the Toolbar button checkboxes for all the apps (without restarting the browser) did.

**Comment by hal90002001** [comment]
I had the same issue.  I fixed it by removing all of the external applications and adding them back one by one. Given how olliebean resolved the issue, though, it looks like I probably could have just made a small change to each one to force a rewrite of the settings instead of having to re-do everything. Oh well.

Now if only I could figure out why the drop-down menu (external application picker/adder) has turned black.
